### PR TITLE
Use Java 8 Stream

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/main/java/org/apache/commons/text/AlphabetConverter.java [hunk 1]
```diff
@@ -104,11 +104,7 @@ private static Integer[] convertCharsToIntegers(final Character[] chars) {
         if (ArrayUtils.isEmpty(chars)) {
             return ArrayUtils.EMPTY_INTEGER_OBJECT_ARRAY;
         }
-        final Integer[] integers = new Integer[chars.length];
-        for (int i = 0; i < chars.length; i++) {
-            integers[i] = (int) chars[i];
-        }
-        return integers;
+        return Arrays.stream(chars).map(aChar -> (int) aChar).toArray(Integer[]::new);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/text/AlphabetConverter.java:107]: You don't need streams @arturobernalg 
Why not just 
```
        final Integer[] integers = new Integer[chars.length];
        Arrays.setAll(integers, i -> (int) chars[i]);
```

File: src/main/java/org/apache/commons/text/similarity/IntersectionSimilarity.java [hunk 2]
```diff
@@ -125,12 +125,7 @@ int uniqueElementSize() {
      * @return The intersection
      */
     private static <T> int getIntersection(final Set<T> setA, final Set<T> setB) {
-        int intersection = 0;
-        for (final T element : setA) {
-            if (setB.contains(element)) {
-                intersection++;
-            }
-        }
+        int intersection = (int) setA.stream().filter(setB::contains).count();
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/text/similarity/IntersectionSimilarity.java:128]: Do we still need the local variable?

Comment (garydgregory) [reply src/main/java/org/apache/commons/text/similarity/IntersectionSimilarity.java:128]: Whichever way we go, we probably need to Javadoc the truncation (PR) or looping of ints (current master).

Comment (arturobernalg) [reply src/main/java/org/apache/commons/text/similarity/IntersectionSimilarity.java]: inline.
meanwhile I have change your remarks.

**Comment by garydgregory** [comment]
The code does not compile @arturobernalg

**Comment by arturobernalg** [comment]
> @arturobernalg there is nothing to review until the builds are green ;-)
Hey @garydgregory 
Yep. looks green now.
TY

**Comment by sergmain** [comment]
Just curious, what is the point to change the code which can be compiled directly to processor commands with code which makes tons of calls with stack usage?
And after implementing 
"[Project Valhalla](https://openjdk.java.net/projects/valhalla/): [Value types](https://en.wikipedia.org/wiki/Value_type), objects without identity but with an efficient memory layout and leading to better results of [escape analysis](https://en.wikipedia.org/wiki/Escape_analysis)".
everything will be reverted back?

**Comment by RockyMM** [comment]
@sergmain in my perspective, streams do bring a lot in readability and maintainability.

Regarding performance, it's difficult to say with guessing. I think we need proper performance testing, but as a layman, I don't expect much difference in performance. Streams are already well optimized in Java and I trust that performance difference will be negligible.

**Pull request closed without merging**